libs/vkd3d/acceleration_structure.c

```diff
@@ -227,6 +244,69 @@ bool vkd3d_acceleration_structure_convert_inputs(const struct d3d12_device *devi
                     RT_TRACE("  AABB stride: %"PRIu64" bytes\n", geom_desc->AABBs.AABBs.StrideInBytes);
                     break;
 
+                case D3D12_RAYTRACING_GEOMETRY_TYPE_OMM_TRIANGLES:
+                    /* Does the runtime validate this as well? */
+                    if (have_aabbs)
+                    {
+                        ERR("Cannot mix and match geometry types in a BLAS.\n");
+                        return false;
+                    }
+                    have_triangles = true;
+
+                    if (!d3d12_device_supports_ray_tracing_tier_1_2(device))
+                    {
+                        ERR("Opacity micromap is not supported.\n");
+                        return false;
+                    }
+
+                    vkd3d_acceleration_structure_convert_triangles(device,
+                            geom_desc->OmmTriangles.pTriangles, &geometry_infos[i], &primitive_count);
+
+                    geometry_infos[i].geometry.triangles.pNext = omm = &omm_infos[i];
+                    omm->sType = VK_STRUCTURE_TYPE_ACCELERATION_STRUCTURE_TRIANGLES_OPACITY_MICROMAP_EXT;
+                    omm->pNext = NULL;
+
+                    switch (geom_desc->OmmTriangles.pOmmLinkage->OpacityMicromapIndexFormat)
+                    {
+                        case DXGI_FORMAT_UNKNOWN:
+                            omm->indexType = VK_INDEX_TYPE_NONE_KHR;
+                            break;
+                        case DXGI_FORMAT_R32_UINT:
+                            omm->indexType = VK_INDEX_TYPE_UINT32;
+                            break;
+                        case DXGI_FORMAT_R16_UINT:
+                            omm->indexType = VK_INDEX_TYPE_UINT16;
+                            break;
+                        case DXGI_FORMAT_R8_UINT:
+                            FIXME_ONCE("Using UINT8 as OMM index format is technically out of spec.\n");
+                            omm->indexType = VK_INDEX_TYPE_UINT8;
```

Comment:
Fwiw this triggers a validation error:

```
Validation Error: [ VUID-VkAccelerationStructureTrianglesOpacityMicromapEXT-indexType-parameter ] | MessageID = 0xd160e0a1
vkGetAccelerationStructureBuildSizesKHR(): pBuildInfo[0].pGeometries[0].geometry.triangles.pNext<VkAccelerationStructureTrianglesOpacityMicromapEXT>.indexType (VK_INDEX_TYPE_UINT8) requires the extensions VK_KHR_index_type_uint8 or VK_EXT_index_type_uint8.
The Vulkan spec states: indexType must be a valid VkIndexType value (https://docs.vulkan.org/spec/latest/chapters/accelstructures.html#VUID-VkAccelerationStructureTrianglesOpacityMicromapEXT-indexType-parameter)
```

Should I enable that extension to silence it?

Reply:
I wouldn't bother right now since this is only optionally enabled.